### ISSUE TITLE
MetadataSubscriber constructing parent class metadata without reflection class

### ISSUE THREAD

**Issue by R3dLantern** [sulu/sulu#6279]
| Q | A
| --- | ---
| Sulu Version | 2.3.5
| PHP Version | 8.0.11
| DB Version | 10.5.8-MariaDB
| Browser Version | Firefox 93.0b9 (64-Bit)

# Actual Behavior

`Sulu\Component\Persistence\EventSubscriber\ORM\MetadataSubscriber` creates `Doctrine\ORM\Mapping\ClassMetadata` instances for parent classes of a given entity, but the instances have no reflection class value:
```php
// Sulu\Component\Persistence\EventSubscriber\ORM\MetadataSubscriber.php:88-98
foreach (\class_parents($metadata->getName()) as $parent) {
    $parentMetadata = new ClassMetadata(
        $parent,
        $configuration->getNamingStrategy()
    );
    // $parentMetadata->getReflectionClass() === null
    // ...
    $configuration->getMetadataDriverImpl()->loadMetadataForClass($parent, $parentMetadata);
}
```

When using PHP8 attributes for Doctrine ORM, `EntityManager::getConfiguration::getMetadataDriverImpl` returns `Doctrine\ORM\Mapping\Driver\AttributeDriver`, whose `loadMetadataForClass` method requires a `$reflClass` field value, else it throws an exception:

```php
// Doctrine\ORM\Mapping\Driver\AttributeDriver.php:58-64
public function loadMetadataForClass($className, ClassMetadata $metadata): void
{
    assert($metadata instanceof ClassMetadataInfo);

    $reflectionClass = $metadata->getReflectionClass();

    $classAttributes = $this->reader->getClassAnnotations($reflectionClass); // throws Exception: Doctrine\ORM\Mapping\Driver\AttributeReader::getClassAnnotations(): Argument #1 ($class) must be of type ReflectionClass, null given, called in /path/to/project/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php on line 64
    
    // ...
}
```

# Expected Behavior

`Sulu\Component\Persistence\EventSubscriber\ORM\MetadataSubscriber` should create parent class metadata with a reflection class instance so that `Doctrine\ORM\Mapping\Driver\AttributeDriver` can properly process class annotations

# Steps to Reproduce

1. Configure Doctrine ORM to use PHP8 attributes:
```yaml
# /path/to/project/config/packages/doctrine.yaml
doctrine:
# ...
  orm:
    # ...
    mappings:
      App:
        # ...
        type: attribute
```
2. Extend `Sulu\Bundle\SecurityBundle\Entity\User` with a class table inheritance entity and create corresponding entities for the descriminator map. Use PHP8 attributes for ORM definition:
```php
#[ORM\Entity]
#[ORM\Table(name: 'se_users')]
#[ORM\InheritanceType('JOINED')]
#[ORM\DiscriminatorColumn(name: 'user_type', type:'string', length: 50)]
#[ORM\DiscriminatorMap([
    'a' => ATypeUser::class,
    'b' => BTypeUser::class,
])]
abstract class User extends SuluUser
{
    // ...
}

#[ORM\Entity]
#[ORM\Table(name: 'proj_user_a')]
class ATypeUser extends User
{
    // ...
}

#[ORM\Entity]
#[ORM\Table(name: 'proj_user_b')]
class BTypeUser extends User
{
    // ...
}
```
3. Perform any activity that involves class metadata handling - in my case it was enough to try to generate a new Doctrine migration file with `php bin/adminconsole doc:mig:diff`.

# Possible Solutions

* `Sulu\Component\Persistence\EventSubscriber\ORM\MetadataSubscriber` could use `Doctrine\ORM\Mapping\ClassMetadataFactory::loadMetadataFor` to create parent class metadata, which already instantiates a `ReflectionClass` field value.

**Comment by niklasnatter** [comment]
Hey, thanks for reaching out!
Using the `ClassMetadataFactory` instead of manually constructing the `ClassMetadata` sounds like the right thing to do 🙂 
Can you create a pull request (targeting the 2.2 branch) for this?

**Comment by niklasnatter** [comment]
Fixed in #6286. Thanks for reporting and fixing this!
